lib/plines/step.rb

```diff
@@ -135,13 +135,20 @@ def jobs_for(batch_data)
       end
     end
 
+    JobBatchCreationAborted = Class.new(StandardError)
+
     def perform(qless_job)
       batch = JobBatch.find(qless_job.client, pipeline,
                             qless_job.data.fetch("_job_batch_id"))
 
       if (retry_delay = batch.paused_retry_delay)
-        qless_job.move(qless_job.queue_name, delay: retry_delay)
-        return
+        if batch.creation_appears_to_be_stuck?
```

Comment:
> Then wouldn't we observe a running, non-stuck job batch that can never complete because it has jobs that are in a failed state, having raised the JobBatchCreationAborted error?

Not quite.   The job batch can never complete without human intervention, but a human could retry this job, and then it could complete fine.

Currently the timeout is an hour, and IMO if it takes a job batch an hour to be created, we've got bigger problems than having to manually retry a job.  We have the timeout set to an hour to avoid the sorts of problems that could occur if the timeout was too "tight".  Your suggestion is a good one in that it would make it more robust and allow us to lower the timeout.  However, I'm not sure it's worth putting effort into it right now, given the urgent prod issues and other sprint work I have on my plate.  This PR makes plines strictly better, IMO, and does not preclude us from implementing the more robust solution at a later time.